include/rpm/rpmpgp.h

```diff
@@ -1113,6 +1119,95 @@ int pgpDigParamsCmp(pgpDigParams p1, pgpDigParams p2);
  */
 unsigned int pgpDigParamsAlgo(pgpDigParams digp, unsigned int algotype);
 
+/** \ingroup rpmpgp
+ * Returns the issuer or the object's Key ID.
+ *
+ * If the object is a signature, then this returns the Key ID stored in the
+ * first Issuer or Issuer Fingerprint subpacket as a hex string.
+ * (This is not authenticated.)
+ *
+ * If the object is a certificate or a subkey, then this returns the key's
+ * Key ID.
+ *
+ * The caller must not free the returned buffer.
+ *
+ * param digp		parameter container
+ * return		an array of PGP_KEYID_LEN bytes.  If the issuer is
+ * 			unknown, this returns an array with all zeros.
+ */
+const uint8_t *pgpDigParamsSignID(pgpDigParams digp);
+
+/** \ingroup rpmpgp
+ * Retrieve the primary User ID, if any.
+ *
+ * Returns the primary User ID, if any.
+ *
+ * If the object is a signature, then this returns NULL.
+ *
+ * If the object is a certificate or a subkey, then this returns the
+ * certificate's primary User ID, if any.
+ *
+ * This interface does not provide a way for the caller to recognize
+ * any embedded NUL characters.
```

Comment:
I’d be fine with just rejecting certificates with user IDs that have embedded NUL characters, on the grounds that this is more likely to be due to an attack than anything else.

Reply:
Although I think your suggestion is reasonable, I think that is out of scope for this change.  (I was just documenting the behavior that I observed.)